### PR TITLE
Add logger.{time,timeEnd} methods

### PR DESCRIPTION
**Pull request by acheronfail**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you use https://nodejs.org/api/perf_hooks.html in the main process?

File: index.js [hunk 1]
```diff
@@ -63,6 +65,31 @@ class Timber {
 		console.error(...args);
 	}
 
+	time(name) {
+		if (name) {
```
Comment (sindresorhus) [inline index.js]: I don’t think it should silently handle no name. Instead it should throw.

File: index.js [hunk 2]
```diff
@@ -17,6 +18,7 @@ class Timber {
 		this.isEnabled = filteredLoggers && options.name ? filteredLoggers.has(options.name) : true;
 		this.name = options.name || '';
 		this._prefixColor = (new Randoma({seed: `${this.name}x`})).color().hex().toString();
+		this._timers = {};
```
Comment (sindresorhus) [inline index.js]: Use a `Map` instead of an object.

File: index.js (hunk 1 above)
Comment (acheronfail) [reply index.js]: I just checked `console.time`, and it nothing is passed, it assigns the name to `"default"`.

I think we should actually copy that, meaning:

```js
logger.time();
logger.timeEnd(); // logs "default: 1234ms"
```

**Comment by acheronfail** [comment]
> Can you use nodejs.org/api/perf_hooks.html in the main process?

`perf_hooks` was added in node `8.5.0` which is recent enough. I've got electron `1.8.2` installed and it doesn't have `perf_hooks` since its node version is only `8.2.1`, that's why I originally just went for this approach. 

I suppose we could add in a `try {} catch {}` to check for perf_hooks and then use that when it's available?

**Comment by sindresorhus** [comment]
Nah, better to wait then. Can you leave a TODO comment about using it when Electron 2.0 is out?

**Comment by sindresorhus** [comment]
Can you add a test and docs?

**Comment by acheronfail** [comment]
Okay, added those. Are they usually flaky? For me the logs seem to come in different orders sometimes.

**Comment by sindresorhus** [comment]
> Okay, added those. Are the tests usually flaky?

No, seems to be new. Would you be able to look into it? It might be because Electron 1.8 was released, and it's now used here, because of https://github.com/sindresorhus/electron-timber/blob/db1998c24b94561e682662c1b754e1e1c1337056/package.json#L45

Just a guess.

**Comment by sindresorhus** [comment]
Can you also fix the merge conflict?

**Comment by acheronfail** [comment]
> No, seems to be new. Would you be able to look into it?

Hmm, running it manually in multiple versions of electron gives consistent results (ie, logging in main & renderer multiple times). 

But whenever I run the tests over and over though sometimes the logs come in a different order (usually logs coming through `log` come before `warn` and `error`) - happens about 1 in 5 tries on my machine.

**Comment by sindresorhus** [comment]
It's really strange that the `main` logs are not always ordered, as they happen synchronously. As a workaround, can you add `.sort()` to both actual and expected arrays and add a short comment on why sort is there?

**Comment by acheronfail** [comment]
Yeah that's a good workaround for now, weird though.

**Comment by sindresorhus** [comment]
> weird though.

That's Electron in a nutshell…

**Comment by sindresorhus** [comment]
This is very good. Thanks for putting so much effort into it. 🙌

**Comment by acheronfail** [comment]
No worries 🎉

**Pull request merged**